crates/cli/src/cli_args.rs

```diff
@@ -0,0 +1,94 @@
+use std::{env, path::PathBuf};
+use store::StoreCommand;
+
+/// Experimental package manager for node.js written in rust.
+#[derive(Debug, Parser)]
+#[clap(name = "pacquet")]
+#[clap(bin_name = "pacquet")]
+#[clap(version = "0.2.1")]
+#[clap(about = "Experimental package manager for node.js")]
+pub struct CliArgs {
+    #[clap(subcommand)]
+    pub command: CliCommand,
+
+    /// Set working directory.
+    #[clap(short = 'C', long, default_value = ".")]
+    pub dir: PathBuf,
+}
+
+#[derive(Subcommand, Debug)]
+pub enum CliCommand {
+    /// Initialize a package.json
+    Init,
+    /// Add a package
+    Add(AddArgs),
+    /// Install packages
+    Install(InstallArgs),
+    /// Runs a package's "test" script, if one was provided.
+    Test,
+    /// Runs a defined package script.
+    Run(RunArgs),
+    /// Runs an arbitrary command specified in the package's start property of its scripts object.
+    Start,
+    /// Managing the package store.
+    #[clap(subcommand)]
+    Store(StoreCommand),
+}
+
+impl CliArgs {
+    /// Execute the command
+    pub async fn run(self) -> miette::Result<()> {
+        let CliArgs { command, dir } = self;
+        let manifest_path = || dir.join("package.json");
+        let npmrc = || Npmrc::current(env::current_dir, home::home_dir, Default::default).leak();
+        let state = || State::init(manifest_path(), npmrc()).wrap_err("initialize the state");
+
+        match command {
+            CliCommand::Init => {
+                PackageManifest::init(&manifest_path()).wrap_err("initialize package.json")?;
+            }
+            CliCommand::Add(args) => args.run(state()?).await?,
+            CliCommand::Install(args) => args.run(state()?).await?,
+            CliCommand::Test => {
+                let manifest = PackageManifest::from_path(manifest_path())
+                    .wrap_err("getting the package.json in current directory")?;
+                if let Some(script) = manifest.script("test", false)? {
+                    execute_shell(script)
+                        .wrap_err(format!("executing command: \"{0}\"", script))?;
+                }
+            }
+            CliCommand::Run(args) => args.run(manifest_path())?,
+            CliCommand::Start => {
+                // Runs an arbitrary command specified in the package's start property of its scripts
+                // object. If no start property is specified on the scripts object, it will attempt to
+                // run node server.js as a default, failing if neither are present.
+                // The intended usage of the property is to specify a command that starts your program.
+                let manifest = PackageManifest::from_path(manifest_path())
+                    .wrap_err("getting the package.json in current directory")?;
+                let command = if let Some(script) = manifest.script("start", true)? {
+                    script
+                } else {
+                    "node server.js"
```

Comment:
This is wrong. We need to check for server.js. Here's the error code from pnpm

```
 ERR_PNPM_NO_SCRIPT_OR_SERVER  Missing script start or file server.js
```

Reply:
Since I didn't write this code and it's not related to the goal of refactoring, I have created an issue ticket: https://github.com/pnpm/pacquet/issues/154